core/templates/dev/head/galleries/gallery.html

```diff
@@ -21,7 +21,12 @@
   <script type="text/javascript">
     GLOBALS.LANGUAGE_CODES_AND_NAMES = JSON.parse(
       '{{LANGUAGE_CODES_AND_NAMES|js_string}}');
-    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse('{{CAROUSEL_SLIDES_CONFIG|js_string}}');
+    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse(
+      '{{CAROUSEL_SLIDES_CONFIG|js_string}}');
+    {% if explorations_list %}
```

Comment:
Instability is mostly in regards to the link not being updated properly
after subsequent searches. The issue is mainly that the searchService
(modified in this branch) is not returning the language codes for the
subsequent searches or the language selections aren't being updated
properly in the front. I'll be out tomorrow evening, but I can chat on
Saturday if you are free.

On Thu, Apr 7, 2016 at 10:52 PM, Sean Lip <notifications@github.com> wrote:

> In core/templates/dev/head/galleries/gallery.html
> <https://github.com/oppia/oppia/pull/1686#discussion_r58982789>:
>
> > @@ -21,7 +21,12 @@
> >    <script type="text/javascript">
> >      GLOBALS.LANGUAGE_CODES_AND_NAMES = JSON.parse(
> >        '{{LANGUAGE_CODES_AND_NAMES|js_string}}');
> > -    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse('{{CAROUSEL_SLIDES_CONFIG|js_string}}');
> > +    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse(
> > +      '{{CAROUSEL_SLIDES_CONFIG|js_string}}');
> > +    {% if explorations_list %}
>
> Thanks for the update! I'm a little concerned about the instability, since
> the aim of the suggestion was to make the overall design simpler. Would you
> like to meet up at some point to chat about the design? It's possible that
> there may be finer points that I've missed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/oppia/oppia/pull/1686/files/f5de98f7a02cee4dce1f7a0f632731856e012de5#r58982789>
>